docs/config.md

```diff
@@ -0,0 +1,68 @@
+# Terramate Configuration
+
+There is a series of different configurations that can be done
+on Terramate, ranging from avoiding duplication to controlling
+order of execution of stacks.
+
+In order to do so, Terramate works with configuration files named
+**terramate.tm.hcl**. These files can be found in any directory
+of a Terramate project, but some configurations will be defined
+only once for an entire project, while others provide merge/overriding
+strategies, while others control only behavior specific for
+stacks.
+
+Here is defined the different kinds of configurations and
+how you can learn more about them. But before getting into the different
+kinds of configurations, what would be a Terramate project?
+
+A Terramate project is essentially any collection of Terraform code
+organized into stacks. You can have all Terraform code together in a single
+directory but that would defeat the purpose of most Terramate features, so
+usually, you have a set of stacks and maybe a set of Terraform modules if they
+are not maintained in a different repository.
+
+It is not a hard requirement for Terramate to work that the project uses git 
+for version control, but features like change detection do depend on git to
+work and will fail if this requirement is not met.
+
+In general, a Terramate project looks like this:
+
+* A git project
+* The git top-level dir is the project base dir
+* Stacks will be organized as different directories
+* Configuration may be present on any dir
+
+# Project Configuration
+
+Per project configuration can be defined only once at the project base dir.
+
+Available project-wide configurations can be found [here](project-config.md)
+
+# Stack Configuration
+
+Before talking about stack specific configuration, lets define what is a
+Terramate stack:
+
+* A dir that has no child dirs (a leaf in the file system tree)
+* Has a valid terramate configuration file (**terramate.tm.hcl**)
+* The terramate configuration file has a `stack {}` block on it
+
+Stacks have configurations that are particular to them, like these:
+
+* [Execution Ordering](execution-order.md)
+
+Even though Terramate metadata is not exclusively about stacks and can be
+used on any hierarchical configuration, it does provide information that can
+be useful to stack configuration, more details [here](metadata.md).
```

Comment:
metadata does only exist in the context of stacks... so if we reference a meta data field in an intermediate or root config, it will be interpreted differently for different stacks...

we could also allow both.... e.g. terramate.stack.path is the path in context of stack.. while terramate.local.path is the current directory  (local is bad naming... could not come up with something better w.g. terramate.dir.path) ... vs. maybe terramate.pwd.path is the current directory (or changed via `--change-dir`) ... all valid information ? wdyt?

Reply:
I was not sure and was leaving each metadata docs to define where they are evaluated, but it does make sense that it will always be evaluated on the context of a stack, will update.